libs/ui/src/button.tsx

```diff
@@ -294,9 +294,18 @@ function disabledStyles(p: ThemedStyledButtonProps): CSSObject {
     };
   }
 
+  const backgroundColor = isInverse(color)
+    ? colors.foreground
+    : colors.background;
+  const foregroundColor = isInverse(color)
+    ? colors.background
+    : colors.foreground;
+
   return {
+    backgroundColor,
```

Comment:
I wonder if we should just make the background transparent? I.e. convert it to `outline` fill mode? Not a huge difference in touch mode since there aren't other colors to layer it over.

Reply:
I think we want to guarantee that the text within the button has the desired contrast against the button background, so I'm hesitant to go with `transparent` on the off chance the button's rendered on top of some other background color - WDYT?

Reply:
I think I've set up the color system roles to guarantee that won't be an issue. Otherwise `outline` fill buttons would always be at risk of having contrast issues. And many of the buttons utilize `outline` fill - it's the default.

But I don't think it's a big deal here either way because in touch mode we pretty much only have one background color that buttons go on (`colors.background`). I don't think we even use the inverse case (though I think it's still good to support it to be safe).